### PR TITLE
[Windows] Focus fails on multiple taps, this closes #249

### LINKED ISSUE #249: [Windows] Focus fails on multiple taps
Tapping preview multiple times causes focus failure and crash on Windows Phone 8.1:
`WinRTError: An operation is currently in progress with the device.`

### PR DESCRIPTION
**Pull request by daserge**

### CODE REVIEW

File: src/windows/BarcodeScannerProxy.js [hunk 1]
```diff
@@ -339,11 +350,31 @@ module.exports = {
...
+                        // This happens on mutliple taps
+                        if (e.number !== OPERATION_IS_IN_PROGRESS) {
+                            console.error('focusAsync failed: ' + e);
+                            error(e);
+                        }
+                        complete();
+                    }
+                }
+
+                capturePreview.addEventListener("playing", onplaying, false);
```
Comment (vladimir-kotikov) [inline src/windows/BarcodeScannerProxy.js]: Can we check for `capturePreview.palying` property before attaching a handler? If it is `true` already we could call `focusAsync` immediately

File: src/windows/BarcodeScannerProxy.js [hunk 2]
```diff
@@ -339,11 +350,31 @@ module.exports = {
...
             focusControl.configure(focusConfig);
-            // Need to wrap this in setTimeout since continuous focus should start only after preview has started. See
-            // 'Remarks' at https://msdn.microsoft.com/en-us/library/windows/apps/windows.media.devices.focuscontrol.configure.aspx
-            return WinJS.Promise.timeout(200)
-            .then(function () {
-                return focusControl.focusAsync();
+
+            // Continuous focus should start only after preview has started. See 'Remarks' at 
+            // https://msdn.microsoft.com/en-us/library/windows/apps/windows.media.devices.focuscontrol.configure.aspx
+            return new WinJS.Promise(function (complete, error, progress) {
```
Comment (vladimir-kotikov) [inline src/windows/BarcodeScannerProxy.js]: nit: `progress` parameter is not used

File: src/windows/BarcodeScannerProxy.js [hunk 3]
```diff
@@ -339,11 +350,31 @@ module.exports = {
...
+                function onplaying() {
+                    capturePreview.removeEventListener("playing", onplaying, false);
+
+                    try {
+                        focusControl.focusAsync().done(function () {
```
Comment (vladimir-kotikov) [inline src/windows/BarcodeScannerProxy.js]: nit: you can just write `focusControl.focusAsync().done(complete, error);
`

File: src/windows/BarcodeScannerProxy.js [hunk 4]
```diff
@@ -339,11 +350,31 @@ module.exports = {
...
-            .then(function () {
-                return focusControl.focusAsync();
+
+            // Continuous focus should start only after preview has started. See 'Remarks' at 
+            // https://msdn.microsoft.com/en-us/library/windows/apps/windows.media.devices.focuscontrol.configure.aspx
+            return new WinJS.Promise(function (complete, error, progress) {
+                function onplaying() {
+                    capturePreview.removeEventListener("playing", onplaying, false);
+
+                    try {
```
Comment (vladimir-kotikov) [inline src/windows/BarcodeScannerProxy.js]: The semantics of `try/catch` block below is very similar to the code you have added to `focus` method. Could you factor this into separate method? Or reuse `focus` method here?

File: src/windows/BarcodeScannerProxy.js (hunk 1 above)
Comment (daserge) [inline src/windows/BarcodeScannerProxy.js]: Addressed, thanks.

**Comment by vladimir-kotikov** [comment]
@daserge, could you please rebase this before merging

**Comment by daserge** [comment]
Reworked and rebased, @vladimir-kotikov could you please take a look once more?

**Comment by vladimir-kotikov** [comment]
LGTM

**Pull request merged**